### PR TITLE
CLIP Guidance and Protobuf Doc

### PR DESCRIPTION
**Pull request by wbrown**
The important change to look at is: `src/proto/generation.proto`

The rest of the changes are regenerations based on the `protobuf` file.

### CODE REVIEW

File: src/proto/generation.proto [hunk 1]
```diff
@@ -97,25 +94,53 @@ message SamplerParameters {
...
   float scaled_step = 1;
   optional SamplerParameters sampler = 2;
   optional ScheduleParameters schedule = 3;
+  optional GuidanceParameters guidance = 4;
+}
+
+// What type of guidance to use.
+enum GuidanceType {
+  GUIDANCE_CFG = 0;               // default is CFG guidance
+  GUIDANCE_CLIP = 1;
```
Comment (Dango233) [inline src/proto/generation.proto:127]: well this is CFG+CLIP actually... a better naming?

**Comment by Dango233** [comment]
Oops

File: src/proto/generation.proto (hunk 1 above)
Comment (wbrown) [reply src/proto/generation.proto]: Changed to `GUIDANCE_CFG_CLIP`.

**Review by arsenetar — APPROVED** [review]

File: src/proto/generation.proto [hunk 2]
```diff
@@ -97,25 +94,53 @@ message SamplerParameters {
...
+  GUIDANCE_PRESET_BALANCED = 2;
+  GUIDANCE_PRESET_QUALITY = 3;
+}
+
+// Parameters that affect the behavior of the guidance, typically used for CLIP.
+// The omission of this field implies the default guidance of CFG.
+message GuidanceParameters {
+  optional GuidanceType type = 1;
+  optional GuidancePreset guidance_preset = 2;
+  optional float guidance_strength = 3;          // 0.0 to 1.0
```
Comment (GusRuss89) [inline src/proto/generation.proto:143]: The comment doesn't seem to match the default value here

Comment (arsenetar) [reply src/proto/generation.proto:143]: The value after the `=` is the field number for the protobuf https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers.

Comment (GusRuss89) [reply src/proto/generation.proto:143]: 🤦 sorry my bad

**Pull request merged**